packages/stripe-integration/src/stripe-upe/stripe-upe-integration-service.spec.ts

```diff
@@ -0,0 +1,667 @@
+                    stripeUPEJsMock,
+                ),
+            ).toBe(false);
+        });
+
+        it('returns false if no payment intent retrieved', async () => {
+            stripeUPEJsMock = {
+                ...getStripeUPEJsMock(),
+                retrievePaymentIntent: jest.fn().mockResolvedValue({
+                    paymentIntent: undefined,
+                }),
+            };
+
+            expect(
+                await stripeUPEIntegrationService.isPaymentCompleted(
+                    StripePaymentMethodType.CreditCard,
+                    stripeUPEJsMock,
+                ),
+            ).toBe(false);
+        });
+
+        it('returns false if payment status not succeed', async () => {
+            stripeUPEJsMock = {
+                ...getStripeUPEJsMock(),
+                retrievePaymentIntent: jest.fn().mockResolvedValue({
+                    paymentIntent: {
+                        status: 'failed',
+                    },
+                }),
+            };
+
+            expect(
+                await stripeUPEIntegrationService.isPaymentCompleted(
+                    StripePaymentMethodType.CreditCard,
+                    stripeUPEJsMock,
+                ),
+            ).toBe(false);
+        });
+    });
+
+    describe('#mapStripePaymentData', () => {
+        beforeEach(() => {
+            jest.spyOn(paymentIntegrationService.getState(), 'getBillingAddress').mockReturnValue(
+                getBillingAddress(),
+            );
+        });
+
+        it('throws error if stipe elements does not initialized', () => {
+            let err;
+
+            try {
+                stripeUPEIntegrationService.mapStripePaymentData(undefined);
+            } catch (error) {
+                err = error;
+            } finally {
+                expect(err).toBeInstanceOf(NotInitializedError);
+            }
+        });
+
+        it('throws error if billing address not defined', () => {
+            let err;
+
+            jest.spyOn(paymentIntegrationService.getState(), 'getBillingAddress').mockReturnValue(
+                undefined,
+            );
+
+            try {
+                stripeUPEIntegrationService.mapStripePaymentData(stripeElementsMock);
+            } catch (error) {
+                err = error;
+            } finally {
+                expect(err).toBeInstanceOf(MissingDataError);
+            }
+        });
+
+        it('throws error if email not defined', () => {
+            let err;
+
+            jest.spyOn(paymentIntegrationService.getState(), 'getBillingAddress').mockReturnValue({
+                ...getBillingAddress(),
+                email: undefined,
+            });
+
+            try {
+                stripeUPEIntegrationService.mapStripePaymentData(stripeElementsMock);
+            } catch (error) {
+                err = error;
+            } finally {
+                expect(err).toBeInstanceOf(MissingDataError);
+            }
```

Comment:
[optional] I am ok with how its done. Just want to say, that expect can be moved to `catch` block and it is not necessary to create `err` variable.

P.s.: If you decide to change that, please check other tests in this file

Reply:
changed